File: remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left = 0, right = 0;
        while(right < nums.length) {
            // count the occurances of the num
            int count = 1;
            while(right + 1 < nums.length && nums[right] == nums[right + 1]) {
                right++;
                count++;
            }

            // replace all the left pointer values with the num
            int frequency = Math.min(count, 2);
            while(frequency > 0) {
                nums[left] = nums[right];
                left++;
                frequency--;
            }

            right++;
        }
        return left;
    }
}

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        size = len(nums)
        if size <= 2: return size

        store = 2
        for i in range(2, size):
            if nums[i] != nums[store - 1] or (nums[i] != nums[store-2]):
                nums[store] = nums[i]
                store += 1
        return store
        